search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Client;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+import org.opensearch.searchpipelines.questionanswering.generative.ext.GenerativeQAParamUtil;
+import org.opensearch.searchpipelines.questionanswering.generative.ext.GenerativeQAParameters;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.ChatCompletionOutput;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.Llm;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.LlmIOUtil;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.ModelLocator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Defines the response processor for generative QA search pipelines.
+ *
+ */
+@Log4j2
+public class GenerativeQAResponseProcessor extends AbstractProcessor implements SearchResponseProcessor {
+
+    // The type of this search processor
+    public static final String SEARCH_PROCESSOR_TYPE = "generative_qa";
+
+    // The model_id of the model registered and deployed in OpenSearch.
+    public static final String CONFIG_NAME_MODEL_ID = "model_id";
+
+    // The name of the model supported by an LLM, e.g. "gpt-3.5" in OpenAI.
+    public static final String CONFIG_NAME_LLM_MODEL = "llm_model";
```

Comment:
We don't need this field, as it is in model meta after registering the remote model with ml-commons.

Reply:
Please see my response above.  This is a per-request parameter.